google/ggrc-core
# GGRC-6928/6927 Allow to turn off on Audit sync of people fields between Assessment and Ticket tracker issue

# Dependencies

This PR is `on hold` until dependencies are merged:
- [x] ~~https://github.com/google/ggrc-core/pull/9129~~

# Solution to be implemented:

AC1: Checkbox should be added to 'New Audit', 'Edit Audit' screens

AC2: Checkbox should appear when user turns ON Ticket Tracker integration. In case turned OFF - no checkbox should be shown.

AC3: Checkbox should be by default checked

AC4: Near checkbox should be a '?' icon. On hover - a help text should be shown. See designs.

AC5: In case checkbox checked - all Audit assessments should sync people fields with corresponding Ticket Tracker fields (Issue reporter, Issue verifier, Issue Assignee, CC list).

AC6: in case checkbox is unchecked: 

AC6.1: Audit captain should stop syncing with Issue reporter, Issue Verifier and CC list for all existing and new Assessments within this Audit

AC6.2: Assessment Assignee should stop syncing with Issue assignee and CC list for all existing and new Assessments within this Audit

AC7: All Assessments that refer to Audit with switched off people sync should have a label near Ticket Tracker section ' Not syncing people'. See designs.

AC8: In case user generates a new Ticket for an Assessment that refer to an Audit with switched off people sync - generate ticket and fill in people fields according to current logic described here. All next background jobs shouldn't sync people fields.

AC9: In case user switches off sync people and then switches it on, people fields should be synced during the next scheduled background job.

# Steps to test the changes

Enable issue-tracker mock by adding following env variables to `docker-compose.override.yml` (or using `export` command):
INTEGRATION_SERVICE_URL=mock
AUTHORIZED_DOMAIN=example.com

Launch the app :`launch_ggrc`

1) Check that checkbox is present on Audit create/edit modal
2) Check that checkbox is present on Audit info pane
3) Check that label is present on Assessment create/edit modal
4) Check that label is present on Assessment info pane

Turn off people sync for audit;
Change audit captains or assessment assignees;
Expected result: no changes in reporter, assignee, verifier, ccs fields in issue tracker.

# Solution description

**FE:**
Add `people_sync_enabled` field to Audit issue-tracker config.
Add sync-people checkbox on Audit info pane and Audit modal.
Add sync-people label on Assessment info pane and Assessment modal.

**BE:**
To implement people sync turn on/off feature the following changes where made:

- New boolean field `people_sync_enabled` is added to `IssuetrackerIssue` model, corresponding migration is created and new alias is added to `IssueTracked`-based mixins.

- Merge methods in assessment sync job (e.g `_merge_reporter`, `_merge_assignee`, `_merge_ccs`) are changes to take into account `people_sync_enabled` flag when merging reporter, assignee and ccs from issue tracker with data from DB.

- New column handler `IssueTrackerCheckboxHandler` is added to allow `people_sync_enabled` change during import.

# Sanity checklist




# PR Review checklist
